lib/model/devicedownloadstate.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2015 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package model
+
+import (
+	"github.com/syncthing/syncthing/lib/protocol"
+	"github.com/syncthing/syncthing/lib/sync"
+)
+
+type deviceFolderFileDownloadState struct {
```

Comment:
I'd love one or two sentences of comments on these types (this one and it's sibling below) on what they actually hold, as an intro the reader. :)